homeassistant/components/hausbus/light.py

```diff
@@ -0,0 +1,306 @@
+    Status as rgbDimmerStatus,
+)
+
+from homeassistant.components.light import (
+    ATTR_BRIGHTNESS,
+    ATTR_HS_COLOR,
+    DOMAIN,
+    ColorMode,
+    LightEntity,
+)
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .config_entry import HausbusConfigEntry
+from .const import ATTR_ON_STATE
+from .device import HausbusDevice
+from .entity import HausbusEntity
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: HausbusConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Haus-Bus lights from a config entry."""
+    # gateway = cast(IEventHandler, hass.data[HAUSBUSDOMAIN][config_entry.entry_id])
+    gateway = config_entry.runtime_data.gateway
+
+    @callback
+    async def async_add_light(channel: HausbusEntity) -> None:
+        """Add light from Haus-Bus."""
+        if isinstance(channel, HausbusLight):
+            async_add_entities([channel])
+
+    gateway.register_platform_add_channel_callback(async_add_light, DOMAIN)
+
+
+class HausbusLight(HausbusEntity, LightEntity):
+    """Representation of a Haus-Bus light."""
+
+    def __init__(
+        self,
+        instance_id: int,
+        device: HausbusDevice,
+        channel: ABusFeature,
+    ) -> None:
+        """Set up light."""
+        super().__init__(channel.__class__.__name__, instance_id, device)
+
+        self._attr_is_on = False
+        self._attr_brightness = 255
+        self._attr_hs_color = (0, 0)
+
+    @staticmethod
+    def is_light_channel(class_id: int) -> bool:
+        """Check if a class_id is a light."""
+        if class_id in (Dimmer.CLASS_ID, RGBDimmer.CLASS_ID, Led.CLASS_ID):
+            return True
+        return False
+
+    def get_hardware_status(self) -> None:
+        """Request status of a light channel from hardware."""
+        raise NotImplementedError
+
+    def set_light_color(self, red: int, green: int, blue: int) -> None:
+        """Set the color of a light channel."""
+        hue, saturation, value = colorsys.rgb_to_hsv(
+            red / 100.0,
+            green / 100.0,
+            blue / 100.0,
+        )
+        params = {
+            ATTR_ON_STATE: True,
+            ATTR_BRIGHTNESS: round(value * 255),
+            ATTR_HS_COLOR: (round(hue * 360), round(saturation * 100)),
+        }
+        self.async_update_callback(**params)
+
+    def set_light_brightness(self, brightness: int) -> None:
+        """Set the brightness of a light channel."""
+        params = {ATTR_ON_STATE: True, ATTR_BRIGHTNESS: (brightness * 255) // 100}
+        self.async_update_callback(**params)
+
+    def light_turn_off(self) -> None:
+        """Turn off a light channel."""
+        params = {
+            ATTR_ON_STATE: False,
+        }
+        self.async_update_callback(**params)
+
+    @staticmethod
+    def handle_light_event(data: Any, channel: HausbusEntity) -> None:
```

Comment:
Why is this a static method instead of an instance method?

Reply:
There is not supposed to be any HausbusLight instance and the method is independent from any instance.

Reply:
I don't understand this explanation.

> There is not supposed to be any HausbusLight instance

Sure, there are no instances of the abstract base class, but there are instances of the concrete classes `HausbusDimmerLight`, `HausbusRGBDimmerLight` and `HausbusLedLight`.

> the method is independent from any instance

This is clearly not the case since we pass in a reference to a `HausbusEntity` which we on the very first line check is a `HausbusLight`?

If this was instead an instance method, we would call it like this from `HausbusGateway`:
```py
        channel = self.get_channel(object_id)
        if channel is not None:
            cast(HausbusLight, channel).handle_light_event(data, channel)
```